File: 백준/Bronze/25304. 영수증/영수증.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split("\n");

let total = Number(input[0]);
let cnt = Number(input[1]);
let sum = 0;
for(let i=2; i<cnt+2; i++) {
    let product = input[i].split(' ');
    sum += Number(product[0])*Number(product[1]);
}

if(total === sum) {
    console.log("Yes");
} else {
    console.log("No");
}

File: 백준/Bronze/25304. 영수증/영수증.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int total = Integer.parseInt(br.readLine());
        int cnt = Integer.parseInt(br.readLine());
        int sum = 0;
        for (int i = 0; i < cnt; i++) {
            String[] product = br.readLine().split(" ");
            sum += Integer.parseInt(product[0]) * Integer.parseInt(product[1]);
        }

        if (total == sum) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }
    }
}